pydantic/class_validators.py

```diff
@@ -12,6 +12,7 @@
 
 
 class Validator:
+    SKIP_ON_FAILURE_KEY = '__validator_skip_on_failure__'  # Applies only to RootValidators
```

Comment:
sorry, this isn't what I meant, I guess I wasn't clear enough.

I meant change `Validator` to take another argument `skip_on_failure`, e.g. after `check_fields`.

Reply:
It doesn't work. if the validator is a function in the line of `    for validator in model.__post_root_validators__:` I can't really access this data..
I can change the whole way this works but I think it's an overkill.

Reply:
it's just a one line change below.

Reply:
Sorry I'm not sure I understood correctly, but my interpretation of your change breaks it 😅

Reply:
Sorry, it was slightly more involved than I had anticipated, I've pushed to your PR to demonstrate what I mean. Hope that's okay and you agree with approach.

Reply:
No problem, looks great. Thank you.